### PR TITLE
REBASED Pr81x l1t-tsg-v9 (part 2) for Sept 2016 MC production

### PR DESCRIPTION
**Pull request by rekovic**
This is the rebased version of #15746.

### CODE REVIEW

**Comment by rekovic** [comment]
+1

**Comment by rekovic** [comment]
please test

**Comment by rekovic** [comment]
The commits 32fb535 to 53752e7 are to addressing code review comments already addressed in #15732, and are relevant here as well.

**Comment by rekovic** [comment]
+1

**Comment by rekovic** [comment]
please test

**Comment by civanch** [comment]
+1

**Comment by davidlange6** [comment]
Hi @rekovic - I made it through the code that is not just white space changes (that code needs some repairs as already described - we can save that for the 80x back port. I think the small changes needed here are:

L1uGTTreeProducer::~L1uGTTreeProducer()
why comment out delete tree_?

remove couts from possibly used in production code 

uint -> unsigned int

remove the using namespace in headers.

**Comment by rekovic** [comment]
Hi @davidlange6 

> L1uGTTreeProducer::~L1uGTTreeProducer()
why comment out delete tree_?

This is not the part of production code of the L1T emulator, but maker of the private Ntuples for L1T studies. It was put in as a temp fix for the problem of Tree being erased. It probably needs to be addressed at some point, but should not be relevant for central CMSSW. 

> remove couts from possibly used in production code

This has already been done here in d99e13f, addressing the exact comment of the big PR
https://github.com/cms-sw/cmssw/pull/15732#discussion_r77598593

> uint -> unsigned int

Addressed in the commit c0fb7b0.

> remove the using namespace in headers

This has already been done here in 650a70e, addressing the exact comment of the big PR
https://github.com/cms-sw/cmssw/pull/15732#discussion_r77605832

**Comment by rekovic** [comment]
+1

**Comment by rekovic** [comment]
please test

File: L1Trigger/L1TNtuples/interface/MuonID.h [hunk 1]
```diff
@@ -9,6 +9,15 @@
...
 #include <DataFormats/PatCandidates/interface/Muon.h>
 
+//vertex
+// EDM formats
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+//local  data formats
+#include "L1Trigger/L1TNtuples/interface/L1AnalysisRecoVertexDataFormat.h"
+
+using namespace muon;
```
Comment (davidlange6) [inline L1Trigger/L1TNtuples/interface/MuonID.h]: @rekovic - these need to  be removed

Comment (rekovic) [inline L1Trigger/L1TNtuples/interface/MuonID.h]: @davidlange6 Now done in a301eac.

**Comment by rekovic** [comment]
+1

**Comment by rekovic** [comment]
please test

**Comment by rekovic** [comment]
please test

**Comment by rekovic** [comment]
please test

**Comment by civanch** [comment]
+1

**Comment by rekovic** [comment]
@slava77 @davidlange6  How do we proceed with the problem with the test above due to missing matrix map?

**Comment by davidlange6** [comment]
@rekovic - not to worry for now - but please start working on the changes needed for the back port...

**Pull request merged**